File: src/main/java/io/github/icusystem/icu_connect/api_icu/IAPIAccount.java
package io.github.icusystem.icu_connect.api_icu;

public interface IAPIAccount {

    void onToken(Token token);
    void onDeviceDetail(DeviceDetail device);
    void onStatus(StatusResponse status);
    void onGetSettings(DeviceSettings settings);
    void onSetSettings();
    void onSetSession();
    void onSetStreamSettings();
    void onGetSession(SessionResponse session);
    void onGetAgeResult(SessionAgeResult session);
    void onGetScanResult(SessionScanResult session);
    void onFaceIDSuccess(EnrollResponse enrollResponse);
    void onFaceIDFail(EnrollError enrollError);
    void onDeleteFaces();
    void onFacesUpdated();
    void onRequestFail(ICUError icuError,String  message);




}

File: src/main/java/io/github/icusystem/icu_connect/api_icu/IAPIAccount.kt
package io.github.icusystem.icu_connect.api_icu;

public interface IAPIAccount {

        fun onToken(token: Token)
        fun onDeviceDetail(device:DeviceDetail)
        fun onStatus(status: StatusResponse)
        fun onGetSettings(settings: DeviceSettings)
        fun onSetSettings()
        fun onSetSession()
        fun onSetStreamSettings()
        fun onGetSession(session:SessionResponse)
        fun onGetAgeResult(session:SessionAgeResult)
        fun onGetScanResult(session:SessionScanResult)
        fun onFaceIDSuccess(enrollResponse: EnrollResponse)
        fun onFaceIDFail(enrollError: EnrollError)
        fun onDeleteFaces()
        fun onFacesUpdated()
        fun onRequestFail(icuError: ICUError, message:String)




}
